### ISSUE TITLE
Break up "respectful" configuration

### ISSUE THREAD

**Issue by brandonchinn178** [fourmolu/fourmolu#281]
Today, `respectful` does three things:
1. Preserve import groups or merge into one import group
2. Omit newlines between top-level declarations if user omitted newlines (or force newlines between)
3. Keep `where` keyword on same line as closing parens in an export list (https://github.com/fourmolu/fourmolu/pull/245)

IMO it's perfectly reasonable for someone to want to respect some of these and not respect others.

@georgefst: What do you think about replacing `respectful: true | false` with separate options for each of these scenarios (and have independent options for future "respectful"-related formatting), e.g.

```yaml
# default: preserve
import-groups: "preserve" | "merge"

# default: false
force-newlines-between-decls: true | false

# default: false
force-where-newline-exports: true | false
```

Pros:
* It makes the options much more understandable
    * "be less aggressive" was always a bit vague to me
* It allows future extendability of any of these options
    * e.g. enabling `import-groups` to autogroup imports like [`eslint-plugin-import`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md) or [`isort`](https://pycqa.github.io/isort/docs/configuration/custom_sections_and_ordering.html)
* It allows users to configure these scenarios independently of each other

Cons:
* Three options is more configuration that just one option
    * Turning off respectfulness now involves turning off three configuration options
* Would be forced to add a separate option for each "respectful"-related configuration, instead of a single catch-all, potentially greatly increasing the number of configuration
    * But any configuration with "respectful" variants would probably want to be extendable anyway (see note about `import-groups` above), so this pattern would encourage us to be more flexible

Related: https://github.com/fourmolu/fourmolu/issues/280

**Comment by georgefst** [comment]
Yeah, I knew we'd have to do this eventually. I think we need to keep a `--respectful` flag which implies all the others (backwards-compatibility, for one thing). This might be awkward to implement - I don't think any of our current flags work this way?

**Comment by brandonchinn178** [comment]
@georgefst why do we need to keep `--respectful`? It's on by default (which all of these options would too), and I've never heard of anyone turning it off. IIUC it was only added to keep parity with ormolu styling (which requires it disabled).

IMO we shouldn't worry about the config being backwards compatible. I would be in favor of erroring if the config file contained any unknown options, though (which would've helped with #306). In the off chance someone manually set `respectful: false`, they'll get an error to update their config, a one time thing when upgrading fourmolu

**Comment by georgefst** [comment]
> IMO we shouldn't worry about the config being backwards compatible.

_In general_, I really think we should. I mentioned something in another thread about how I'd even like to provide migration scripts.

But it's a very good point that the existence of `--respectful` is kind of a historical curiosity, which very few people are likely to have manually disabled. So I'm willing to make an exception here.

> I would be in favor of erroring if the config file contained any unknown options, though (which would've helped with https://github.com/fourmolu/fourmolu/issues/306).

Agreed.

**Comment by georgefst** [comment]
> But it's a very good point that the existence of `--respectful` is kind of a historical curiosity, which very few people are likely to have manually disabled. So I'm willing to make an exception here.

I think I've changed my mind on this, unfortunately. I've seen repos that have copy-pasted one of the config files from this repo, so they will have an explicit `respectful` setting.

**Comment by brandonchinn178** [comment]
If we kept `respectful` for backwards compatibility, would you be amenable to showing a deprecation warning and removing it X versions later, or would you be opposed to any removal?
